rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/CompactView.java

```diff
@@ -166,269 +165,107 @@ private void changeSelection(String name) {
-                                nt.refresh(n);
-                            }
-                        }
-                    }
+                if (nodeSource.isChanged()) {
+                    flow.redrawNodeSource(nodeSource);
                 }
             }
+        }
+    }
 
-            /*
-             * now remove the difference between the nodes from this method call,
-             * and the ones from the previous call
-             */
-            for (Entry<String, NodeSource> oldNs : this.oldNodes.entrySet()) {
-                /* Keep NS */
-                NodeSource newNs = nodes.get(oldNs.getKey());
-                if (newNs != null) {
-                    for (Entry<String, Node> oldDepl : oldNs.getValue().getDeploying().entrySet()) {
-                        /* Keep deploying Node */
-                        if (newNs.getDeploying().containsKey(oldDepl.getKey())) {
+    private void initializePanel() {
+        normalFlow = new CompactFlowPanel();
+        myNodesFlow = new CompactFlowPanelOwn();
 
-                        }
-                        /* Deploying Node to be removed */
-                        else if (curTiles.contains(oldDepl.getKey())) {
-                            removeTile(oldDepl.getKey());
-                        }
-                    }
+        normalFlowCanvas = new WidgetCanvas(normalFlow);
+        myNodesFlowCanvas = new WidgetCanvas(myNodesFlow);
 
-                    for (Entry<String, Host> oldHost : oldNs.getValue().getHosts().entrySet()) {
-                        /* Keep host */
-                        Host newHost = newNs.getHosts().get(oldHost.getKey());
-                        if (newHost != null && newHost.getNodes().size() > 0) {
-                            for (Entry<String, Node> oldNode : oldHost.getValue().getNodes().entrySet()) {
-                                /* Keep node */
-                                if (newHost.getNodes().containsKey(oldNode.getKey())) {
-                                }
-                                /* Node to be removed */
-                                else if (curTiles.contains(oldNode.getKey())) {
-                                    removeTile(oldNode.getKey());
-                                }
-                            }
+        root.addMember(normalFlowCanvas);
 
-                        }
-                        /* Host to be removed */
-                        else if (curTiles.contains(oldHost.getValue().getId())) {
-                            removeTile(oldHost.getValue().getId());
-                            for (String n : oldHost.getValue().getNodes().keySet()) {
-                                removeTile(n);
-                            }
-                        }
-                    }
+        root.addResizedHandler(event -> {
+            int w = root.getWidth();
+            int h = root.getHeight();
+            normalFlow.setPixelSize(w - root.getScrollbarSize(), h - root.getScrollbarSize());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)